runtimes/core/src/metrics/manager.rs

```diff
@@ -0,0 +1,173 @@
+use crate::{
+    encore::runtime::v1::{self as pb, Environment},
+    metadata::{process_env_substitution, ContainerMetadata},
+    metrics::{
+        exporter::{self, Exporter},
+        registry::Registry,
+    },
+};
+use std::sync::Arc;
+use std::time::Duration;
+
+#[derive(Debug, Clone)]
+enum ProviderType {
+    Gcp(pb::metrics_provider::GcpCloudMonitoring),
+    EncoreCloud(pb::metrics_provider::GcpCloudMonitoring),
+    // TODO(fredr) add these:
+    // - Aws(pb::metrics_provider::AwsCloudWatch),
+    // - Prometheus(pb::metrics_provider::PrometheusRemoteWrite),
+    // - Datadog(pb::metrics_provider::Datadog),
+}
+
+impl ProviderType {
+    fn from_config(provider: &pb::MetricsProvider) -> Option<Self> {
+        match &provider.provider {
+            Some(pb::metrics_provider::Provider::Gcp(config)) => Some(Self::Gcp(config.clone())),
+            Some(pb::metrics_provider::Provider::EncoreCloud(config)) => {
+                Some(Self::EncoreCloud(config.clone()))
+            }
+            _ => {
+                log::warn!("unsupported metrics provider: {:?}", provider.provider);
+                None
+            }
+        }
+    }
+
+    fn create_exporter(
+        &self,
+        env: &Environment,
+        http_client: &reqwest::Client,
+        runtime_handle: tokio::runtime::Handle,
+    ) -> Arc<dyn Exporter + Send + Sync> {
+        match self {
+            Self::Gcp(config) | Self::EncoreCloud(config) => {
+                runtime_handle.block_on(Self::create_gcp_exporter(config, env, http_client))
```

Comment:
A bit annoying to need to block here. I guess the only reason we need to do that is because the container metadata needs to be collected? Can we do that lazily/in the background, similar to the lazy gcp client?